erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -431,7 +431,7 @@ def update_args_for_serial_no(serial_no_doc, serial_no, args, is_new=False):
 		"via_stock_ledger": args.get("via_stock_ledger") or True,
 		"supplier": args.get("supplier"),
 		"location": args.get("location"),
-		"sales_order": frappe.db.get_value(args.get("voucher_type"), args.get("voucher_no"), "sales_order_no"),
+		"sales_order":  frappe.db.get_value(args.get("voucher_type"), args.get("voucher_no"), "sales_order_no") if args.get("voucher_type") == "Stock Entry" else "",
```

Comment:
@vishdha I'm not sure about this change, is this proper?

Reply:
@ShahzadNaser What is the purpose to add "sales_order" parameter here.
if you check this validate_reserved_serial_no_consumption which is added for reserve purpose that means it can not randomly assign serial number

and another function def update_so_in_serial_number(self): which updates the sales order number in the serial number when stock entry purpose is manufacturing.

What is purpose to add sales_order parameter

Reply:
@visdha  I doesn't remember exact case but this triggered on SLE entery and `sales_order` param already passed. I just add condition to check `sales_order_no` key only if SLE triggered via stock entery because `sales_order_no` key only present in that specific doctype. But as you know SLE triggered from various operations and same applies with `update_args_for_serial_no` because this is triggering on SLE.